file.go

```diff
@@ -71,17 +73,16 @@ func (fi *File) Open(flags Flags) (_ FileDescriptor, _retErr error) {
 		return nil, fmt.Errorf("file opened for neither reading nor writing")
 	}
 
-	fi.nodeLock.RLock()
-	node := fi.node
-	fi.nodeLock.RUnlock()
+	node, err := fi.GetNode()
+	if err != nil {
+		return nil, err
+	}
 
-	// TODO: Move this `switch` logic outside (maybe even
-	// to another package, this seems like a job of UnixFS),
-	// `NewDagModifier` uses the IPLD node, we're not
+	// TODO: Move this logic outside (maybe even to another package, this seems
+	// like a job of UnixFS), `NewDagModifier` uses the IPLD node, we're not
```

Comment:
I think the only reason we're doing this is because we don't support symlinks... We should just open the file then check if it's a symlink.

Reply:
There is no logic change here. Just a reformat of an old comment.

Reply:
@Stebalien This TODO, and the TODO below it, will probably disappear when we move to use an IPLD-prime node.  So, will leave this as it for now.